apollo-router/src/plugins/external.rs

```diff
@@ -12,41 +11,173 @@ use bytes::Bytes;
 use http::header::HeaderName;
 use http::HeaderMap;
 use http::HeaderValue;
-use hyper::body;
+use hyper::client::HttpConnector;
 use hyper::Body;
+use hyper_rustls::ConfigBuilderExt;
+use hyper_rustls::HttpsConnector;
 use schemars::JsonSchema;
-use serde::de::DeserializeOwned;
 use serde::Deserialize;
 use serde::Serialize;
+use tower::timeout::TimeoutLayer;
 use tower::util::MapFutureLayer;
 use tower::BoxError;
+use tower::Service;
 use tower::ServiceBuilder;
 use tower::ServiceExt;
 
 use crate::error::Error;
+use crate::error::LicenseError;
 use crate::layers::async_checkpoint::AsyncCheckpointLayer;
 use crate::layers::ServiceBuilderExt;
 use crate::plugin::Plugin;
 use crate::plugin::PluginInit;
 use crate::register_plugin;
+use crate::services::apollo_graph_reference;
+use crate::services::execution;
 use crate::services::external::Control;
 use crate::services::external::Externalizable;
 use crate::services::external::PipelineStep;
+use crate::services::external::DEFAULT_EXTERNALIZATION_TIMEOUT;
+use crate::services::external::EXTERNALIZABLE_VERSION;
 use crate::services::router;
-use crate::Context;
+use crate::services::subgraph;
+use crate::services::supergraph;
+use crate::tracer::TraceId;
 
 pub(crate) const EXTERNAL_SPAN_NAME: &str = "external_plugin";
 
+type HTTPClientService = tower::timeout::Timeout<hyper::Client<HttpsConnector<HttpConnector>>>;
+
+#[async_trait::async_trait]
+impl Plugin for ExternalPlugin<HTTPClientService> {
+    type Config = Conf;
+
+    async fn new(init: PluginInit<Self::Config>) -> Result<Self, BoxError> {
+        if init.config.stages != Default::default() {
+            apollo_graph_reference().ok_or(LicenseError::MissingGraphReference)?;
+        }
+
+        let mut http_connector = HttpConnector::new();
```

Comment:
NIT: we keep make HTTP clients (uplink, jwks, etc), it might be good to factor those so we have one client with safe defaults (like timeouts). And ideally, I'd like it to have limited support for caching

Reply:
I'm opening a followup